src/textual/_compositor.py

```diff
@@ -877,16 +877,14 @@ def cuts(self) -> list[list[int]]:
             return self._cuts
 
         width, height = self.size
-        screen_region = self.size.region
         cuts = [[0, width] for _ in range(height)]
 
         intersection = Region.intersection
         extend = list.extend
 
         for region, clip in self.visible_widgets.values():
-            region = intersection(region, clip)
-            if region and (region in screen_region):
-                x, y, region_width, region_height = region
+            x, y, region_width, region_height = intersection(region, clip)
```

Comment:
We don't need the `region in screen_region` because `self.visible_widgets` has already been limited to visible widgets.